src/server/pps/server/pipeline_controller.go

```diff
@@ -149,11 +149,9 @@ func (pc *pipelineController) Start(timestamp time.Time) {
 		case <-pc.ctx.Done():
 			return
 		case ts := <-pc.bumpChan:
-			err := pc.step(ts)
-			// we've given up on the step, check if the error indicated that the pipeline should fail
-			if err != nil {
-				log.Errorf("PPS master: error starting a pipelineController for pipeline '%s': %v", pc.pipeline,
-					errors.Wrapf(err, "failing pipeline %q", pc.pipeline))
+			// pc.step returns true if the pipeline was deleted, and the controller can try to shutdown
+			if isDelete, _ := pc.step(ts); isDelete {
```

Comment:
Without having traced the code too carefully, it seems like it might still make sense to log any errors here, since they're still returned? Wdyt?

Reply:
I suggested returning this error and then having the caller not read it in an earlier comment 🤦‍♀️. Feel free to ignore (I guess I do think the error should be logged